### ISSUE TITLE
Error message "Cannot load a translation for the language: ay" after copying mmexini.db3 from earlier version

### ISSUE THREAD

**Issue by tactilis** [moneymanagerex/moneymanagerex#4470]
MMEX version:
 - [x] 1.5.15 Beta 3 

Operating System:
 - [x] Windows
 - [ ] Mac OSX
 - [ ] Linux 

# Description of the bug

The error message _"Cannot load a translation for the language: ay"_ was displayed after copying an existing mmexini.db3 from 1.5.14  to 1.5.15 Beta 3 and starting MMEX.

# Reproduction

Is the bug reproducible? 
 - [x] Always 
 - [ ] Randomly 
 - [ ] Happened only once

Reproduction steps:

I have two MMEX installations:

**MAIN**: MMEX 1.5.14 (via the .exe installer)
**BETA**:  1.5.15 Beta 3, commit ID 6d909ec6 (portable installation)

I ran BETA for the first time, opened my existing testing mmb and noted that the toolbar icons and Home page scaling were different from what I was used to with MAIN:

![image](https://user-images.githubusercontent.com/4060092/170510751-c217a6f1-2361-4df3-bafa-0c610125dd6e.png)

 
 
![image](https://user-images.githubusercontent.com/4060092/170510792-f936d475-4b8e-45a1-9c8a-4b9758377a34.png)

 
  
This is because:
- MAIN uses the mmexini.db3 in C:\Users\Martin\AppData\Roaming\MoneyManagerEx
- BETA comes with its own (new) mmexini.db3 in the folder that I unzipped the files to.

![image](https://user-images.githubusercontent.com/4060092/170505355-efe8ddbc-faf2-48f1-9801-0d807a40d9c1.png)
 
I closed BETA, deleted its mmexini.db3, and copied MAIN's mmexini.db3 to BETA's folder.

I ran BETA again and immediately got this error message:

![image](https://user-images.githubusercontent.com/4060092/170512075-632f74a6-f82e-49f7-9755-687f5f440f38.png)

I clicked OK and the MMEX start screen appeared. I opened the testing .mmb and checked the language:

![image](https://user-images.githubusercontent.com/4060092/170515572-eced27d4-bbf5-49a4-945c-98c8903b33e4.png)

 
 
![image](https://user-images.githubusercontent.com/4060092/170516838-a13fcf6c-710a-4bb2-a3b9-cdc285b7d15e.png)

Note that my Windows system language is English (United Kingdom). 

**Expected result:**

Running 1.5.15 Beta 3 with the mmexini.db3 from 1.5.14:

a) should not result in a _"Cannot load a translation for the language: ay"_ error message

b) should have the same application language set as was used in 1.5.14

I am concerned that an incompatible change may have been made for 1.5.15 and that there may be upgrade implications (or at least user confusion) when people move from 1.5.14 (or earlier) to 1.5.15.

**Comment by whalley** [comment]
@tactilis  This https://github.com/moneymanagerex/moneymanagerex/issues/4296 perhaps

**Comment by tactilis** [comment]
> This https://github.com/moneymanagerex/moneymanagerex/issues/4296 perhaps

It seems to be related to that issue - but has a different error message.

IMO, the solution suggested in [https://github.com/moneymanagerex/moneymanagerex/issues/4296#issuecomment-1120222529](https://github.com/moneymanagerex/moneymanagerex/issues/4296#issuecomment-1120222529)...

> "Solution: you just need to reassign the language when you first launch a new version of the application."

...does not seem a good one for the several thousand(?) of users of MMEX when they upgrade from 1.5.14 (or older) to 1.5.15.

Is there no mechanism within MMEX to:

a) apply upgrade scripts when a new version of MMEX is first run?

and/or

b) Detect that the mmexini.db3 being opened was written by a pre-1.5.15 version of MMEX and make the necessary changes when 1.5.15 is run?

**Comment by vomikan** [comment]
It depends of wxWidgets version.
@whalley it should be noted that we use different versions of wx for mac and windows. And this will be a problem for users who switch from one OS to another.
We need to roll back wx.

**Comment by PMaff** [comment]
> It depends of wxWidgets version. @whalley it should be noted that we use different versions of wx for mac and windows. And this will be a problem for users who switch from one OS to another. We need to roll back wx.

Roll back wxWidget to 3.1.5 for the Linux versions also?

**Comment by vomikan** [comment]
> Roll back wxWidget to 3.1.5 for the Linux versions also?

Yes, please.

**Comment by PMaff** [comment]
For future versions only or do we also need a replacement of existing Linux versions?

The following existing 1.5.14 versions are affected:
mmex_1.5.14-1~jammy_amd64.deb 
mmex-1.5.14-1.x86_64.rpm - Suse Tumbleweed

**Comment by vomikan** [comment]
> For future versions only or do we also need a replacement of existing Linux versions?

Better for all versions.

**Comment by PMaff** [comment]
@vomikan  I recompiled wxWidget 3.1.5 on Suse Tumbleweed and I was able to compile mmex 1.5.14 with that but just of curiosity I tested a bit and found the error shown in the attached short film.

The error with the comment that stays there is already solved by you in a later version.

But the error that the complete display vanishes was new for me (I never use money tips).
This is also on Ubuntu 18.04 LTS.
Do you know that display vanishing already or shall I open a new issue ticket?


https://user-images.githubusercontent.com/93879362/170833742-f3656ee7-0317-4227-98ff-0f54df4d8d53.mp4

**Comment by vomikan** [comment]
@PMaff It's LINUX!!! Do as you like!

**Comment by PMaff** [comment]
> @PMaff It's LINUX!!! Do as you like!

I just tried 

Version: 1.5.15-Beta.3 64-bit
• Build on May 28 2022 18:45:40
• Database version: 7 • (aes128cbc)
• Git commit: bef96ad50 (2022-05-28)
• Git branch: master

with wxWidgets 3.1.5 (wxGTK 3.24)
and there everything is ok, so no need for a bug issue.
;-)

**Comment by vomikan** [comment]
> only the right side of the display is erased.

Is it depend on wx version?

**Comment by PMaff** [comment]
> > only the right side of the display is erased.
> 
> Is it depend on wx version?

Since both versions of mmex use the same wxWidget version (3.1.5) I think it is rather dependent on changes
that were done in 1.5.15-Beta.3 64-bit and not dependent on wxWidget version.
Question is, if erasing the right side of the display is the desired action?
Shall I try to make a screen video from both versions?

**Comment by vomikan** [comment]
I suppose it could be this fix.

https://github.com/vomikan/moneymanagerex/commit/95fa2415c20c54c5f975f170670dbaaa28e52ca8

![image](https://user-images.githubusercontent.com/6836805/170873111-1645e864-92b2-4341-9d3b-d17111ffee6f.png)

Apart from the massive renaming of functions, in fact, only two lines have been deleted.
```m_nav_tree_ctrl->SetEvtHandlerEnabled(false);```
```m_nav_tree_ctrl->SetEvtHandlerEnabled(true);```

**Comment by PMaff** [comment]
I think the handling in 1.5.15-Beta.3 is ok.

**Comment by vomikan** [comment]
> I think the handling in 1.5.15-Beta.3 is ok.

I am talking about 1.5.14

**Comment by PMaff** [comment]
Maybe we should move all this into a different issue ticket?

> > I think the handling in 1.5.15-Beta.3 is ok.
> 
> I am talking about 1.5.14

1.5.14 has the false handling, which is shown in the video above.
But since this is released we cannot do anything about it.

For the newer 1.5.15/1.5.16: if we want to be more sophisticaed then keep the view of 
the current selected transaction if you switch to "Show Money Tips":
if there is a note for the transaction, show the note,
if there is no note, either display a "money tip" or the empty note field.
Changing the view always shows a money or tool tip if both are selected, which is ok.

**Comment by tactilis** [comment]
@PMaff

> Maybe we should move all this into a different issue ticket?

I think so, as you have moved this issue far away from the original topic, which was:

   **Error message "Cannot load a translation for the language: ay" after copying mmexini.db3 from earlier version**

**Comment by PMaff** [comment]
https://github.com/moneymanagerex/moneymanagerex/issues/4494
